src/components/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,305 @@
+      <Dialog open={openCreateDialog} onClose={handleCancelCreate}>
+        <DialogTitle>{DASHBOARD.CREATE_DIALOG.TITLE}</DialogTitle>
+        <DialogContent>
+          {columnData?.map((col: any) => {
+            const { field, headerName, type } = col;
+            if (field === "actions") return null; 
+            const value = newRowData[field] || "";
+
+            return (
+              <TextField
+                key={field}
+                label={headerName}
+                name={field}
+                type={type || "text"}
+                value={value}
+                onChange={handleCreateFieldChange}
+                fullWidth
+                margin="normal"
+              />
+            );
+          })}
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleCancelCreate} color="primary">
+            {DASHBOARD.CREATE_DIALOG.CANCEL}
+          </Button>
+          <Button
+            onClick={handleSaveCreate}
+            color="primary"
+            disabled={!isFormValid} 
+          >
+            {DASHBOARD.CREATE_DIALOG.CREATE}
+          </Button>
+        </DialogActions>
+      </Dialog>
+
+      {/* Edit Dialog */}
+      <Dialog open={editingRow !== null} onClose={handleCancelEdit}>
+        <DialogTitle>{DASHBOARD.EDIT_DIALOG.TITLE}</DialogTitle>
+        <DialogContent>
+          {columnData?.map((col: any) => {
+            const { field, headerName, type } = col;
+            if (field === "actions") return null; 
+            const value = editedRowData[field] || "";
+
+            return (
+              <TextField
+                key={field}
+                label={headerName}
+                name={field}
+                type={type || "text"}
+                value={value}
+                onChange={handleFieldChange}
+                fullWidth
+                margin="normal"
+              />
+            );
+          })}
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleCancelEdit} color="primary">
+            {DASHBOARD.EDIT_DIALOG.CANCEL}
+          </Button>
+          <Button
+            onClick={handleSaveEdit}
+            color="primary"
+            disabled={!isEditFormValid} 
+          >
+            {DASHBOARD.EDIT_DIALOG.SAVE}
+          </Button>
+        </DialogActions>
+      </Dialog>
+
+      {/* Delete Dialog */}
+      <Dialog open={openDeleteDialog} onClose={cancelDelete}>
+        <DialogTitle>{DASHBOARD.DELETE_DIALOG.TITLE}</DialogTitle>
+        <DialogContent>
+          <p>{DASHBOARD.DELETE_DIALOG.BODY}</p>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={cancelDelete} color="primary">
+            {DASHBOARD.DELETE_DIALOG.CANCEL}
+          </Button>
+          <Button onClick={confirmDelete} color="secondary" variant="contained">
+            {DASHBOARD.DELETE_DIALOG.DELETE}
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </>
+  );
+};
+
```

Comment:
some thing on this component too big please split it to more component/hooks. 
300 lines is a lot